pyaedt/modules/solutions.py

```diff
@@ -1849,30 +1847,29 @@ def plot_2d_cut(
                 return False
             curves.append([x, y, "{}={}".format(y_key, data[y_key][theta_idx])])
 
-        if show:
-            show_legend = True
-            if len(curves) > 15:
-                show_legend = False
-            if is_polar:
-                return plot_polar_chart(
-                    curves,
-                    xlabel=x_key,
-                    ylabel=quantity,
-                    title=title,
-                    snapshot_path=image_path,
-                    show_legend=show_legend,
-                )
-            else:
-                return plot_2d_chart(
-                    curves,
-                    xlabel=x_key,
-                    ylabel=quantity,
-                    title=title,
-                    snapshot_path=image_path,
-                    show_legend=show_legend,
-                )
+        show_legend = show
```

Comment:
show_legend is different from show so I don't think show_legend should be set to show. They are two different things.

Reply:
I do agree and, as a matter of fact, I just wanted to follow previous logic (if show then show_legend = True).
I'll update it such that they are different things